javasource/mxawsiot/impl/MqttConnector.java

```diff
@@ -189,14 +189,14 @@ public boolean isSubscribed(String topic) {
         public void subscribe(String topic, String onMessageMicroflow) throws MqttException {
             logger.info("MqttConnection.subscribe");
             try {
+                subscriptions.put(topic, new MqttSubscription(topic, onMessageMicroflow));
```

Comment:
All subscriptions need to be registered before the connection can be made, otherwise you run a risk of loosing messages.

Start listening on the connection should be an explicit microflow action.